modular_skyrat/modules/customization/modules/surgery/organs/silicon_alt_brains.dm

```diff
@@ -0,0 +1,70 @@
+	begin_activation_message = "<span class='notice'>You carefully locate the manual activation switch and start the compact AI circuit's boot process.</span>"
+	success_message = "<span class='notice'>The compact AI circuit pings, and its lights start flashing. Success!</span>"
+	fail_message = "<span class='notice'>The compact AI circuit buzzes quietly, and the golden lights fade away. Perhaps you could try again?</span>"
+	new_mob_message = "<span class='notice'>The compact AI circuit chimes quietly.</span>"
+	recharge_message = "<span class='warning'>The compact AI circuit isn't ready to activate again yet! Give it some time to recharge.</span>"
+
+/obj/item/organ/internal/brain/ipc_positron/circuit
+	name = "compact AI circuit"
+	desc = "A compact and extremely complex circuit, perfectly dimensioned to fit in the same slot as a synthetic-compatible positronic brain. It is usually slotted into the chest of synthetic crewmembers."
+	icon = 'modular_skyrat/master_files/icons/obj/alt_silicon_brains.dmi'
+	icon_state = "circuit-occupied"
+	inhand_icon_state = "electronic"
+	lefthand_file = 'icons/mob/inhands/items/devices_lefthand.dmi'
+	righthand_file = 'icons/mob/inhands/items/devices_righthand.dmi'
+
+/obj/item/organ/internal/brain/ipc_positron/mmi
+	name = "compact man-machine interface"
+	desc = "A compact man-machine interface, perfectly dimensioned to fit in the same slot as a synthetic-compatible positronic brain. Unfortunately, the brain seems to be permanently attached to the circuitry, and it seems relatively sensitive to it's environment. It is usually slotted into the chest of synthetic crewmembers."
+	icon = 'modular_skyrat/master_files/icons/obj/surgery.dmi'
+	icon_state = "mmi-ipc"
+
+// CODE THAT ACTUALLY APPLIES THE BRAINS.
+// See modular_skyrat/master_files/code/modules/client/preferences/brain.dm for Synth/IPC application.
+
+/proc/prefs_get_brain_to_use(value, is_cyborg = FALSE)
+	switch(value)
+		if(ORGAN_PREF_POSI_BRAIN)
+			return is_cyborg ? /obj/item/mmi/posibrain : /obj/item/organ/internal/brain/ipc_positron
+		if(ORGAN_PREF_MMI_BRAIN)
+			return is_cyborg ? /obj/item/mmi : /obj/item/organ/internal/brain/ipc_positron/mmi
+		if(ORGAN_PREF_CIRCUIT_BRAIN)
+			return is_cyborg ? /obj/item/mmi/posibrain/circuit : /obj/item/organ/internal/brain/ipc_positron/circuit
+
+/mob/living/silicon/robot/Initialize(mapload)
+	. = ..()
+	// Intentionally set like this, because people have different lore for their cyborgs, and there's no real non-invasive way to print posibrains that match.
+	RegisterSignal(src, COMSIG_MOB_MIND_TRANSFERRED_INTO, .proc/update_brain_type)
+
+/// Sets the MMI type for a cyborg, if applicable.
+/mob/living/silicon/robot/proc/update_brain_type()
+	var/obj/item/mmi/new_mmi = prefs_get_brain_to_use(client?.prefs?.read_preference(/datum/preference/choiced/brain_type), TRUE)
+	if(!new_mmi || new_mmi == mmi.type)
+		return
+	new_mmi = new new_mmi()
```

Comment:
Would this make anyone borged into the chosen brain type?

Reply:
Yes, and is semi intended, as is said in the PR body. This can be changed to not do that if wanted, though the code will be relatively shit.